src/sidebars/favourites/geometry-service.js

```diff
@@ -0,0 +1,63 @@
+(function() {
+    'use strict';
+
+    angular.module('openfin.geometry', [])
+        .factory('geometryService', [function() {
+
+            var Point = function(aX, aY) {
+                this.x = aX || 0;
+                this.y = aY || 0;
+            };
+            Point.constructor = Point;
+
+            var Rectangle = function(rect) {
+                this.origin = new Point(rect.left, rect.top);
+                this.extent = new Point(rect.width, rect.height);
+            };
+            Rectangle.constructor = Rectangle;
+
+            Rectangle.prototype = {
```

Comment:
This came mostly from the OpenFin tutorial on tearouts, but I guess your point stands as that's just passing it up the chain. Don't mind really, as long as it stays working (which I'm pretty sure it is now).